### PR TITLE
Update Litecoin P2SH

### PR DESCRIPTION
**Pull request by losh11**
### Litecoin Core 0.13.3 (reference client)

https://github.com/litecoin-project/litecoin/blob/master/src/chainparams.cpp#L129

### LoafWallet (iOS Wallet)

https://github.com/litecoin-association/LoafWallet/blob/master/BreadWallet/NSString%2BBitcoin.h#L30

### CODE REVIEW

**Comment by dcousens** [comment]
Do you have a citation or reference?
And can you provide some test fixtures maybe?

**Comment by losh11** [comment]
Sorry, I was about to do that, and got disturbed by stuff.

Sure, I can fix tests, but all tests seem to pass on my machine `2795 passing (12s)`.

**Comment by dcousens** [comment]
@losh11 then we need tests! :+1:

File: src/networks.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ module.exports = {
       private: 0x019d9cfe
     },
     pubKeyHash: 0x30,
-    scriptHash: 0x05,
+    scriptHash: 0x50,
```
Comment (junderw) [inline src/networks.js]: `0x32` ?

The link in litecoin is base-10

**Comment by dcousens** [comment]
Not going to merge until we have some example Litecoin P2SH addresses added to tests to ensure the constant is actually valid...

**Comment by dcousens** [comment]
@junderw @losh11 if either of you have example Litecoin P2SH addresses... please either update this PR,  submit a different PR, or just post them.

**Comment by afk11** [comment]
Someone grab a LTC P2SH address and run base58 decode on it?

@losh11 as mentioned, the constant should be expressed as 0x32 or 50, but not 0x50.

**Comment by afk11** [comment]
Guessing everyones happy, whoever second passes this please merge?

**Comment by dcousens** [comment]
If someone just adds fixtures... We're good to merge.

**Comment by afk11** [comment]
will do!

**Comment by losh11** [comment]
That's great guys! Sorry got really busy, so I didn't notice this...

**Comment by dcousens** [comment]
Published as `3.0.3`

**Comment by rplopes** [comment]
This change makes it impossible to work with legacy P2SH litecoin addresses, which will still show up in live systems.

Is there a recommended approach for dealing with such valid legacy transaction versions? I'm guessing adding a `legacyScriptHash` property to all networks might not be the best long term strategy. Adding a `litecoinLegacy` network wouldn't work, as a transaction builder only accepts one network, and a transaction can mix the old and new addresses.

If it's something that is not possible to do, would it be something worthy to build support for?

**Pull request merged**